### PR TITLE
Confidence

### PR DESCRIPTION
**Pull request by colehurwitz**
Naive implementation of confidence over 2*sigma pixels. On edges, it will repeat edge values rather than exceed image limits.

### CODE REVIEW

**Comment by colehurwitz** [comment]
Leaving the old method for confidence commented out for now

**Comment by colehurwitz** [comment]
Does this logic make sense?

File: lightning_pose/data/utils.py [hunk 1]
```diff
@@ -284,15 +284,40 @@ def generate_heatmaps(
...
-    k = locs[:, :, None, 1, None].type(torch.int64)  # y first
-    l = locs[:, :, 0, None, None].type(torch.int64)  # x second
-    vals = heatmaps[i, j, k, l].squeeze(-1).squeeze(-1)  # get rid of singleton dims
-    return vals
+    k = locs[:, :, None, 1, None].type(torch.int64)
+    l = locs[:, :, 0, None, None].type(torch.int64)
+    pix_to_consider = int(np.ceil(sigma * 2.0)) #get all pixels within two stds.
+    offsets = list(np.arange(-pix_to_consider, pix_to_consider+1))
+    vals_all = []
+    for offset in offsets:
```
Comment (danbider) [inline lightning_pose/data/utils.py:314]: looks good, height and width may differ (i.e., we may have a non square heatmap). so the max clamping should be `heatmap_height-1` then `heatmap_width-1`. you currently use `max=heatmaps.shape[2]-1` for both. other than that, fine

**Review by danbider — CHANGES_REQUESTED** [review]
can also leave a bit more of an explanation in comments

File: lightning_pose/data/utils.py (hunk 1 above)
Comment (colehurwitz) [reply lightning_pose/data/utils.py:316]: Fixed, thanks!

**Comment by colehurwitz** [comment]
I think my formatter kicked in.. sorry

**Review by danbider — APPROVED** [review]
great

**Pull request merged**